source/common/buffer/buffer_impl.h

```diff
@@ -119,6 +120,7 @@ class Slice {
       reservable_ = rhs.reservable_;
       drain_trackers_ = std::move(rhs.drain_trackers_);
       account_ = std::move(rhs.account_);
+      releasor_.swap(rhs.releasor_);
```

Comment:
I don't think `swap` is correct in this case. Putting the old releasor into `rhs` is incorrect, isn't it?

Reply:
You are right, let me fix that